### PR TITLE
Remove dead links and unused TOC

### PR DESCRIPTION
**Pull request by Licenser**
remove manual TOC from rfc page (we have a generated one now) and change some dead links to point to the right place

### CODE REVIEW

**Review by mfelsche — DISMISSED** [review]
It is approval day today

File: rfc/index.md [hunk 1]
```diff
@@ -94,7 +77,7 @@ beforehand, to ascertain that the RFC may be desirable; having a consistent
 impact on the project requires concerted effort toward consensus-building.
 
 The most common preparations for writing and submitting an RFC include talking
-the idea over on our [Tremor chat](https://join.slack.com/t/tremor-debs/shared_invite/enQtOTMxNzY3NDg0MjI2LTQ4MTU4NjlkZDk0MmJmNmIwYjU0ZDc1OTNjMGRmNzUwZTdlZGVkMWFmNGFkZTAwOWJlYjlkMDZkNGNiMjQ2NzI), and occasionally posting "pre-RFCs" on the
+the idea over on our [Tremor chat](https://chat.tremor.rs), and occasionally posting "pre-RFCs" on the
```
Comment (mfelsche) [inline rfc/index.md:80]: :+1: good catch

**Review by mfelsche — APPROVED** [review]

File: rfc/index.md (hunk 1 above)
Comment (Licenser) [reply rfc/index.md:80]: :D ja I found a few more too, so much easier now that everything is in one place

**Review by mfelsche — APPROVED** [review]

**Comment by skoech** [comment]
Nice!

**Pull request merged**